services/user-management/src/main/java/com/foodie/user/controllers/RolePermissionController.java

```diff
@@ -0,0 +1,183 @@
+        this.roleService = roleService;
+        this.permissionService = permissionService;
+        this.rolePermissionService = rolePermissionService;
+    }
+
+    private Role validateAndGetRole(UUID roleId) {
+        return roleService.findByIdAndDeletedAtIsNull(roleId).orElse(null);
+    }
+
+    private Permission validateAndGetPermission(UUID permitId) {
+        return permissionService.findByIdAndDeletedAtIsNull(permitId).orElse(null);
+    }
+
+    @PostMapping(produces = "application/json", consumes = "application/json")
+    public ResponseEntity<FoodieBaseResponse> createRolePermission(@Valid @PathVariable("role") UUID roleId,
+            @Valid @RequestBody String[] permissionIds) {
+
+        if (permissionIds == null || permissionIds.length == 0) {
+            return ResponseEntity.badRequest().body(new FoodieBaseResponse("Permission cannot be null"));
+        }
+
+        Role role = validateAndGetRole(roleId);
+        if (role == null) {
+            return ResponseEntity.badRequest().body(new FoodieBaseResponse("Role not found."));
+        }
+
+        List<RolePermission> rolePermissions = new ArrayList<>();
+
+        for (String permitId : permissionIds) {
+            Permission permission = validateAndGetPermission(UUID.fromString(permitId));
+            if (permission == null) {
+                return ResponseEntity.badRequest().body(new FoodieBaseResponse("Permission not found."));
+            }
+            rolePermissions.add(new RolePermission(role, permission));
+        }
+
+        log.info("Creating Role permissions");
+        List<RolePermission> createdPermissions = rolePermissionService.saveAll(rolePermissions);
+        return ResponseEntity.status(HttpStatus.CREATED).body(new FoodieBaseResponse(createdPermissions));
+    }
+
+    @GetMapping(path = "/{id}", produces = "application/json")
+    public ResponseEntity<FoodieBaseResponse> getRolePermissionById(@Valid @PathVariable("id") UUID id) {
+
+        log.info("Fetching Role permission with ID: {}", id);
+
+        Optional<RolePermission> permission = rolePermissionService.findByIdAndDeletedAtIsNull(id);
+
+        return permission.map(value -> ResponseEntity.ok().body(new FoodieBaseResponse(value)))
+                .orElseGet(() -> ResponseEntity.notFound().build());
+    }
+
+    @GetMapping(produces = "application/json")
+    public ResponseEntity<FoodieBaseResponse> getAllPermissionsForRole(@Valid @PathVariable("role") UUID roleId,
+            @RequestParam(defaultValue = "0") int page,
+            @RequestParam(defaultValue = "10") int size,
+            @RequestParam(defaultValue = "id") String sort) {
+
+        log.info("Fetching all Role {} permissions with page={}, size={}, sort={}", roleId, page, size, sort);
+
+        Pageable pageable = PageRequest.of(page, size, Sort.by(sort));
+        Page<RolePermission> permissions = rolePermissionService.findAllByRoleIdAndDeletedAtIsNull(roleId, pageable);
```

Comment:
What happens if record not found here?

Reply:
An error is thrown and catch

Reply:
So this method will return a 500 error or another error code?